mod/homepage/lib/javascript/script_homepage_carousel.js.coffee

```diff
@@ -135,8 +149,32 @@ isScrolledIntoView = (elem) ->
   elemBottom = elemTop + $(elem).height();
   ((elemBottom <= docViewBottom) && (elemTop >= docViewTop));
 
-animation = (elem) ->
-  # animation CountUp.js
+runAnimation = (elem) ->
+  options = { useEasing: true, useGrouping: true, separator: ',', decimal: '.', };
+  animationNumber = new CountUp($(elem).attr('id'),  0, parseInt($(elem).text()), 0, 3.5, options);
+  animationNumber.start()
+
+controlAnimate = (elem) ->
+  numberElementsControls.push( {id: $(elem).attr('id'), animated: false} )
+
+# has this element been animated?
+isAnimate = (id) ->
+  aux = false
+  numberElementsControls.forEach (elem) ->
+    if elem.id == id && elem.animated 
+      aux = true 
+      return
+  aux  
+
+# this element is already animated now
+nowIsAnimate = (id) ->
```

Comment:
wait.  what's the difference between this and isAnimated?

maybe the other should be `isAnimatable` and this is `isAnimated`?

sorry, I was confused before.

Reply:
this function only sets the element as animated

Reply:
isAnimated is to know if an element has already been animated

Reply:
oh, now that I'm looking at the functionality, I see that it's not a boolean call.  (And I should note that even if it returned true/false, usually a boolean call should not have side effects)

this should be something like `animateElem`, no